internal/driver/webui.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package driver
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/google/pprof/internal/graph"
+	"github.com/google/pprof/internal/plugin"
+	"github.com/google/pprof/internal/report"
+	"github.com/google/pprof/profile"
+)
+
+// webUI holds the state needed for serving a browser based interface.
+type webUI struct {
+	prof    *profile.Profile
+	options *plugin.Options
```

Comment:
The motivation behind the UI field in plugin.Options is that it would allow errors to be redirected to an interactive UI instead of always printing to stderr.

Would it be possible to redefine options.UI so that it would show such errors somewhere in the UI (either a pop-up or maybe in red in a corner)?

For instance, if you type a function name that doesn't exist and click "focus" we get a print on stderr saying "Focus expression matched no samples". It would be better if that showed up on the browser.